Start documenting working with Masu for nise. (#192)
project-koku/nise: docs/working_with_masu.rst

===========
Working with Masu
===========
~~~~~
About
~~~~~

Nise is a developer tool for generating sample cost and use data for testing purposes. Although nise can generate standalone reports, its most common use case is to create reports to be consumed by `koku's <https://github.com/project-koku/koku>`_ python module masu. This section provides a walkthrough for how to use nise along with masu to generate providers in koku.

OCP Providers
=============
In order to create any ocp providers with masu, nise will need to place the csv files in the masu expected location. Masu expects the files to be placed under koku's :code:`testing/pvc_dir/insights_local` `directory <https://github.com/project-koku/koku/tree/master/testing/pvc_dir>`_. If :code:`insights_local` does not exist yet you will have to create it.

Basic OCP provider
------------------
**Step One: The Nise Command**

> To create a basic OCP provider you will have to provide three key peices of information.

Reviewer: ```suggestion
To create and ingest OCP sample data, there are three required pieces of information..
```

   1. :code:`--ocp-cluster-id $(cluster_id)` <string> Ex. my-cluster_id

>    2. :code:`--insights-upload testing/pvc_dir/insights_local` Masu Expected location

Reviewer: I find `Masu Expected location` to be a bit confusing. How does `Ingestion upload path` sound? Or maybe `Report upload path`?

>    3. :code:`--static-report-file $(srf_yaml)` Path to a static report `file <https://github.com/project-koku/nise/blob/master/example_aws_static_data.yml>`_

Reviewer: In this preamble portion, I think we need to add additional detail around why we're using a static file, what it does for us, and why we can't generate ocp-on-aws data without it.

Author: Added this sentence in the ocp-on-aws section: `Which is why we are using report files because the autogentated data option does not have the ability to keep the resouce_id the same across AWS and OCP sample data.`

.. highlight:: bash

::

  nise --ocp --ocp-cluster-id $(cluster_id) --insights-upload testing/pvc_dir/insights_local --static-report-file $(srf_yaml)
.. highlight:: none

**Step Two: Create the Provider**

> After creating the csvs under the :code:`insights_local` directory, the provider will then need to be added by posting to the provider endpoint.

Reviewer: ```suggestion
Once Nise has placed the sample data into the :code:`insights-upload` directory, a provider will need to be added to Koku. To accomplish this, we will use the Koku REST API to create a new provider.
```

.. highlight:: bash

::

   curl -d '{"name": "OCP_PROVIDER_NAME", "type": "OCP", "authentication": {"provider_resource_name": "$(cluster_id)"}}' -H "Content-Type: application/json" -X POST http://0.0.0.0:8000/api/cost-management/v1/providers/

.. highlight:: none

> - The :code:`$(cluster_id)` must match the cluster_id provided in the nise command.

Reviewer: ```suggestion
- The :code:`$(cluster_id)` must match the cluster_id provided to Nise.
```

> **Step Three: Trigger Masu**

Reviewer: ```suggestion
**Step Three: Initiate data ingestion**
```

> You can trigger Masu by hitting the download endpoint: http://127.0.0.1:5000/api/cost-management/v1/download/

Reviewer: ```suggestion
There is an endpoint used for development and testing that is not intended for use in production environments. To manually trigger a data ingestion task, you can access this endpoint on the Masu micro-service by sending an HTTP GET request to: :code:`http://YOUR_KOKU_SERVICE_HOSTNAME/KOKU_API_PREFIX/v1/download/`
```

OCP on AWS Provider
-------------------

> In order to create an OCP on AWS provider, you must first create an ocp provider and then an aws provider. Each provider will need to be created from a static report file, and the resource_id and dates in both AWS and OCP static report files match.

Reviewer: ```suggestion
To ingest sample OCP-on-AWS data, there are multiple steps involved. First create and ingest OCP data as described in the previous section. Then sample AWS data must be generated and ingested. Each sample data set must be created using a separate static report file. The resource_id and dates in both AWS and OCP static report files must match.
```

**Step One: Create the OCP provider**

> The steps to create the ocp provider will be the exact same as the `Basic OCP provider`_ steps above, except you must use the static file under the :code:`ocp_on_aws` example `here <https://github.com/project-koku/nise/blob/master/examples/ocp_on_aws/ocp_static_data.yml>`_.

Reviewer: ```suggestion
Generate and ingest sample OCP data as described in the `Create and Ingest OCP Sample Data`_ section above. Use the example static file provided :code:`ocp_on_aws` example `here <https://github.com/project-koku/nise/blob/master/examples/ocp_on_aws/ocp_static_data.yml>`_.
```

> **Step Two: The AWS nise command**

Reviewer: My suggestions for this section mirrors a lot of what I suggested above in the OCP section. So, apply those suggestions to this section of the doc, too.

The main goal for the doc should be to have a consistent set of terms that are used throughout, and to have consistent punctuation and grammar (e.g. Nise, not nise; CSV not csv). Also, try to aim for short, precise sentences. The more complex the statement, the more difficult it is for the reader to understand your meaning.
